workflow/docker: github matrix array to deal with docker images

hjpotter92 opened the pull request:
Reduces multiple workflow actions to a single manageable manifest.

hjpotter92 committed:
workflow/docker: Test github matrix array to deal with docker images

hjpotter92 committed:
docker.yaml: yaml and its single/double quotes!

hjpotter92 committed:
docker.yaml: Fix typo in dockerfile name

hjpotter92 committed:
docker.yaml: Do not notify for PRs/custom branches

victorges commented on .github/workflows/docker.yaml:
```diff
@@ -0,0 +1,77 @@
+        with:
+          push: true
+          build-args: |
+            version=${{ steps.tags.outputs.version }}
```
Ooh, so we need the tags step only for the version build arg now?

hjpotter92 replied:
yeah! i couldn't find an alternative to this yet :(

hjpotter92 replied:
yeah! i couldn't find an alternative to this yet :(

victorges reviewed (commented):
Looks awesome! Just a remark on the tag filter since I think we might wanna keep it as "notify every time" if that doesn't create any problems.

hjpotter92 commented on .github/workflows/docker.yaml:
```diff
@@ -0,0 +1,77 @@
+          {name: 'record-tester', file:'Dockerfile.record-tester'},
+          {name: 'orch-tester', file:'Dockerfile.orch-tester'},
+          {name: 'mist-api-connector', file:'Dockerfile.mist api-connector'},
+          {name: 'streamtester', file:'Dockerfile'},
```
yep. it generates a set of 5 separate tests

hjpotter92 commented on .github/workflows/docker.yaml:
```diff
@@ -73,5 +75,12 @@ jobs:
+
       - name: Notify that new build has been uploaded
+        if: ${{ steps.match-tag.outputs.match != '' }}
         run: curl -X POST https://holy-bread-207a.livepeer.workers.dev
```
i wasn't sure how the worker notifications work, and assumed that the current docker tag might get deployed, which is not wanted in all cases.

> just have it be a no-op most of the time since the updated images don't matter to the currently deployed infra.

that makes it easier. i can remove the `match-tag` entirely.

hjpotter92 committed:
docker.yaml: Update github actions to newer version.

 - Add a dependabot daily checker for outdated github actions
 - Remove `match-tag` step from action
 - Send notifications to cloudflare worker on each image generation

victorges reviewed (approved):
LGTM!
